src/test/java/org/mskcc/cmo/metadb/CmoLabelGeneratorServiceTest.java

```diff
@@ -0,0 +1,169 @@
+     * Expected behavior: Should return a label with
+     *     incremented sample count and new cmoPatientId
+     * @throws JsonProcessingException
+     * @throws JsonMappingException
+     * @throws CloneNotSupportedException 
+     * @throws IllegalArgumentException 
+     */
+    @Test
+    public void testCmoLabelGenForSampleWithPatientCorrection()
+            throws JsonMappingException, JsonProcessingException, IllegalArgumentException, CloneNotSupportedException {
+        MockJsonTestData requestJson = mockedRequestJsonDataMap
+                .get("mockPublishedRequest1JsonDataWith2T2N");
+        // Parse requestMap and sampleList from mockPublishedRequest1JsonDataWith2T2N json
+        Map<String, String> mappedRequestJson = mapper.readValue(
+                requestJson.getJsonString(), Map.class);
+        SampleMetadata[] samples = mapper.convertValue(mappedRequestJson.get("samples"),
+                SampleMetadata[].class);
+
+        // existing sample for patientId: C-MP789JR
+        List<SampleMetadata> existingSamples = new ArrayList<>();
+        existingSamples.add(samples[0]);
+        existingSamples.add(samples[1]);
+
+        // updated SampleMetadata
+        SampleMetadata updatedSample = mapper.convertValue(
+                samples[0].clone(), SampleMetadata.class);
+        updatedSample.setCmoPatientId("C-newPatient");
+        updatedSample.setPrimaryId("newIgoId");
+
+        // generate cmoLabel for sample with updates
+        String newCmoLabel = cmoLabelGeneratorService.generateCmoSampleLabel(
+                updatedSample, existingSamples);
+
+        Assert.assertEquals("C-newPatient-X003-d02", newCmoLabel);
+
+    }
+
+    /**
+     * testCmoLabelGenForExistingSampleWithPatientCorrection
+     * Tests if a new cmoLabel is properly generated for
+     *     a sample with metadata with updated cmoPatientId and primaryId/igoId
+     * Expected behavior: Should return a label with
+     *     only a new cmoPatientId change
+     * @throws JsonProcessingException
+     * @throws JsonMappingException
+     * @throws CloneNotSupportedException 
+     * @throws IllegalArgumentException 
+     */
+    @Test
+    public void testCmoLabelGenForExistingSampleWithPatientCorrection()
+            throws JsonMappingException, JsonProcessingException, IllegalArgumentException, CloneNotSupportedException {
+        MockJsonTestData requestJson = mockedRequestJsonDataMap
+                .get("mockPublishedRequest1JsonDataWith2T2N");
+        // Parse requestMap and sampleList from mockPublishedRequest1JsonDataWith2T2N json
+        Map<String, String> mappedRequestJson = mapper.readValue(
+                requestJson.getJsonString(), Map.class);
+        SampleMetadata[] samples = mapper.convertValue(mappedRequestJson.get("samples"),
+                SampleMetadata[].class);
+
+        // existing sample for patientId: C-MP789JR
+        List<SampleMetadata> existingSamples = new ArrayList<>();
```

Comment:
can you add a comment with the existing/matching sample's current cmo label before the update happens?